### ISSUE TITLE
Some insights on coAST

### ISSUE THREAD

**Issue by tushar-rishav** [coala-analyzer/coAST#2]
It'd be helpful if we get to know more about what problems coAST would solve? 
cc @sils1297

**Comment by sils1297** [comment]
well the idea is I can have any programming language, transform it to coAST and then I have a lot of algorithms that operate on coAST and do cool code analysis. I only need to write the algorithms once for all languages then, right?

Would you mind clarifying it in the repo a bit?

**Comment by tushar-rishav** [comment]
@sils1297  Sure thing!
So that's cool. What sort of algorithms you are expecting to run on AST? I assume language specifics checks won't apply here. We can run something which is more general. Maybe things like Cyclomatic complexity? Is there anything else in particular?

**Comment by tushar-rishav** [comment]
@sils1297  I'd summarize what we discuss here and update the idea page in the end? Probably, I'd have better content then

**Comment by sils1297** [comment]
nothing in particular, if coast is good it can be rather complicated things
too

Sincerely,

Lasse Schuirmann

contact@viperdev.io
http://viperdev.io/

2016-03-14 22:35 GMT+01:00 Tushar <notifications@github.com>:

**Comment by Makman2** [comment]
@tushar-rishav you are aware that coAST is mainly for having a language-independent AST generation?

**Comment by tushar-rishav** [comment]
@sils1297 @Makman2  IIUC we need to create a rule/grammar to create a parser that works for all languages? *If* we want coAST to create AST for all the languages. One rule that fits all?

**Comment by sils1297** [comment]
I think there'll be no way around making a parser for every language. Do
you know https://github.com/BNFC/bnfc ? I talked to the maintainer briefly
last guadec, that could be a possible thing...

Sincerely,

Lasse Schuirmann

contact@viperdev.io
http://viperdev.io/

2016-03-18 14:36 GMT+01:00 Tushar <notifications@github.com>:

**Comment by tushar-rishav** [comment]
@sils1297  True! It's difficult task. I was going through https://github.com/orlandohill/waxeye . They seem to parse C, Java, Javascript, Python, Ruby , Scheme. So once we have AST we can do analysis on it.

**Comment by Makman2** [comment]
Performance is also something we need to consider. Writing a parser in pure python could be fairly slow^^ Using Flex and Bison is in fact the right approach for this kind of work imo^^

**Comment by Makman2** [comment]
(If we write them ourselves)
